### DISCUSSION TITLE
How to add gravity to airflow FEM simulation

### DISCUSSION THREAD

**Discussion by brandon-barclay** [idaholab/moose#21495] [Q&A General] [answered]
Hello, I am attempting to use MOOSE to simulate radionuclide deposition with FEM, however, I am struggling to add a kernel to represent gravitational settling of the particle through the domain before it reaches the boundary. I have tried to index the gradients for the Z gradient component however the residual contribution seems to be 0. It appears that using vector variables / kernels would make this a simpler task however all of my kernels are for non vector variables so I am hoping there is a possible workaround. 

Thanks for reading and for your assistance.

**Comment by GiudGiud** [comment, marked as answer]
Hello

What does the equation look like for this problem? And which term is the gravitational settling?
Vector variables/kernels are never required, they're only done for simplicity or efficiency

Happy 4th,
Guillaume

**Reply by brandon-barclay** [reply]
I have attached the weak form of the equation I am attempting to model, with the deposition term highlighted. 

Thank you for your interest in helping me solve this problem. 

<img width="718" alt="Screen Shot 2022-07-05 at 12 57 37 PM" src="https://user-images.githubusercontent.com/91636862/177378897-d0cc5367-c109-48bd-99fe-c13d7d1908f6.png">

**Reply by GiudGiud** [reply]
so what do you have for computeQpResidual currently?
I would see something like:
```
-_nu_s[_qp] * _test[_i][_qp] * _grad_C[2][_qp]
```
nu_s being a material property and _grad_C being obtained using the coupledGradient method

**Reply by brandon-barclay** [reply]
This is almost exactly what I had tried to use, however, I get an error stating 
```
error: type 'const libMesh::VectorValue<MetaPhysicL::DualNumber<double, MetaPhysicL::SemiDynamicSparseNumberArray<double, unsigned long long, MetaPhysicL::NWrapper<53>>, true>>' does not provide a subscript operator
  return _settling_velocity[_qp] * _test[_i][_qp] * _grad_u[2][_qp];
```

**Reply by GiudGiud** [reply]
ok so probably the [2] is wrong. maybe it's (2) or maybe it s at a different position

**Reply by brandon-barclay** [reply]
I was able to get it to work with `_grad_u[_qp](2)` - thanks for your help!

**Reply by GiudGiud** [reply]
ah weird. I ll try to remember. I m used to grad_phi[_j][_qp] and grad_test[_i][_qp], didnt know it was flipped for the variables

**Reply by lindsayad** [reply]
`grad_phi` and `grad_test` are two dimensional arrays of 3-vectors (gradients) indexed by shape function, quadrature point, and finally spatial dimension. `grad_u` is a one-dimensional array of 3-vectors (gradients) without the shape function index, so indexed first by quadrature point and then by spatial dimension
